homeassistant/components/music_assistant/__init__.py

```diff
@@ -0,0 +1,164 @@
+"""Music Assistant (music-assistant.io) integration."""
+
+from __future__ import annotations
+
+import asyncio
+from typing import TYPE_CHECKING
+
+from music_assistant.client import MusicAssistantClient
+from music_assistant.client.exceptions import CannotConnect, InvalidServerVersion
+from music_assistant.common.models.enums import EventType
+from music_assistant.common.models.errors import MusicAssistantError
+
+from homeassistant.config_entries import ConfigEntry, ConfigEntryState
+from homeassistant.const import CONF_URL, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import Event, HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.issue_registry import (
+    IssueSeverity,
+    async_create_issue,
+    async_delete_issue,
+)
+
+from .const import DOMAIN, LOGGER
+from .helpers import MassEntryData
+
+if TYPE_CHECKING:
+    from music_assistant.common.models.event import MassEvent
+
+PLATFORMS = ("media_player",)
+
+CONNECT_TIMEOUT = 10
+LISTEN_READY_TIMEOUT = 30
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up from a config entry."""
+    # ruff: noqa: PLR0915
+    http_session = async_get_clientsession(hass, verify_ssl=False)
+    mass_url = entry.data[CONF_URL]
+    mass = MusicAssistantClient(mass_url, http_session)
+
+    try:
+        async with asyncio.timeout(CONNECT_TIMEOUT):
+            await mass.connect()
+    except (TimeoutError, CannotConnect) as err:
+        raise ConfigEntryNotReady(
+            f"Failed to connect to music assistant server {mass_url}"
+        ) from err
+    except InvalidServerVersion as err:
+        async_create_issue(
+            hass,
+            DOMAIN,
+            "invalid_server_version",
+            is_fixable=False,
+            severity=IssueSeverity.ERROR,
+            translation_key="invalid_server_version",
+        )
+        raise ConfigEntryNotReady(f"Invalid server version: {err}") from err
+    except Exception as err:
+        LOGGER.exception("Failed to connect to music assistant server", exc_info=err)
+        raise ConfigEntryNotReady(
+            f"Unknown error connecting to the Music Assistant server {mass_url}"
+        ) from err
```

Comment:
We only allow catching bare exceptions in the config flow, we should be more specific here